src/lib/visualization/abstract_visualizer.hpp

```diff
@@ -244,27 +244,43 @@ class AbstractVisualizer {
 
   std::string _wrap_label(const std::string& label) {
     if (label.length() <= MAX_LABEL_WIDTH) return label;
-
-    // Split by word so we don't break a line in the middle of a word
-    std::vector<std::string> label_words;
-    boost::split(label_words, label, boost::is_any_of(" "));
-
-    std::stringstream wrapped_label;
-    auto current_line_length = 0;
-
-    for (const auto& word : label_words) {
-      auto word_length = word.length() + 1;  // include whitespace
-
-      if (current_line_length + word_length > MAX_LABEL_WIDTH) {
-        wrapped_label << "\\n";
-        current_line_length = 0u;
+    std::stringstream label_stream;
+
+    // 1. Split label into lines
+    std::vector<std::string> lines;
+    boost::split(lines, label, boost::is_any_of("\n"));
+
+    for (size_t line_idx = 0; line_idx < lines.size(); ++line_idx) {
+      if (line_idx > 0) label_stream << '\n';
+      const auto& line = lines.at(line_idx);
```

Comment:
It should be safe to use `[]` instead of `.at()` to avoid bounds checking. However, since visualization is not performance critical, it should not be an issue.